tests/e2e/tests/pilot/util/environment.go

```diff
@@ -163,21 +170,58 @@ func (e *Environment) ToTemplateData() TemplateData {
 		CABundle:               e.CABundle,
 		RDSv2:                  e.Config.RDSv2,
 		ImagePullPolicy:        e.Config.ImagePullPolicy,
+		MultiCluster:           e.Config.ClusterRegistriesDir,
 	}
 }
 
 // Setup creates the k8s environment and deploys the test apps
 func (e *Environment) Setup() error {
-	if e.Config.KubeConfig == "" {
-		e.Config.KubeConfig = "pilot/pkg/kube/config"
-		log.Info("Using linked in kube config. Set KUBECONFIG env before running the test.")
-	}
 	var err error
-	if _, e.KubeClient, err = kube.CreateInterface(e.Config.KubeConfig); err != nil {
-		return err
+	if e.Config.KubeConfig[0] != "" {
+		if _, e.KubeClient[0], err = kube.CreateInterface(e.Config.KubeConfig[0]); err != nil {
+			return err
+		}
+        }
+	// CLusterRegistiresDir indicates the Kubernetes cluster config should come from files versus KUBECONFIG environment
+	// Variable.  The test config can be defined to use either a single cluster or 2 clusters
+	if e.Config.ClusterRegistriesDir != "" {
+		e.clusterStore, err = clusterregistry.ReadClusters(e.Config.ClusterRegistriesDir)
+		if e.clusterStore == nil {
+			return err
```

Comment:
I take it that just returning here w/ no messaging is OK. i.e. the calling function will provide enough info on the failure in Setup().

Reply:
I can add a log message.